src/xmipp/libraries/reconstruction/subtract_projection.h

```diff
@@ -0,0 +1,114 @@
+/***************************************************************************
+ *
+ * Authors:    Estrella Fernandez Gimenez (me.fernandez@cnb.csic.es)
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+ #ifndef _PROG_SUBTRACT_PROJECTION
+ #define _PROG_SUBTRACT_PROJECTION
+
+ #include "core/metadata_vec.h"
+ #include "core/xmipp_program.h"
+ #include "core/xmipp_image.h"
+ #include "data/fourier_filter.h"
+ #include "data/fourier_projection.h"
+
+
+ class ProgSubtractProjection: public XmippProgram
+ {
+ private:
+    /** Filename of the reference volume */
+    FileName fnVolR;
+    FileName fnParticles;
```

Comment:
that's a lot of attributes. 
Many of them are however used just in specific methods, so preferably declare them directly there